lms/services/vitalsource/service.py

```diff
@@ -187,15 +183,17 @@ def get_user_reference(self, lti_params: LTIParams) -> str | None:
 
         return value
 
-    def h_license_check(self, lti_user: LTIUser, lti_params: LTIParams) -> None:
+    def has_h_license(self, lti_user: LTIUser, lti_params: LTIParams) -> bool:
         """Check if the user of the current launch has a license for the H LTI app."""
         if not self._student_pay_enabled or lti_user.is_instructor:
             # Not a school using student pay or the current user it's an instructor.
-            return
+            return True
 
         user_reference = self.get_user_reference(lti_params)
         if not self._sso_client.get_user_book_license(user_reference, self.H_SKU):
-            raise VitalSourceStudentPayNoLicense()
+            return False
+
+        return True
```

Comment:
I think we could get rid of the third branch in this method by doing this:

```python
def has_h_license(self, lti_user: LTIUser, lti_params: LTIParams) -> bool:
  """Check if the user of the current launch has a license for the H LTI app."""
  if not self._student_pay_enabled or lti_user.is_instructor:
    # Not a school using student pay or the current user it's an instructor.
    return True

  user_reference = self.get_user_reference(lti_params)
  return self._sso_client.get_user_book_license(user_reference, self.H_SKU)
```

Reply:
Yup :+1:, wrapped it on a `bool()`